File: Sources/App/Controllers/AlbumFilter.swift
//
//  AlbumFilter.swift
//  FinchServer
//
//  Created by Stefan Klein Nulent on 23/03/2025.
//

import Vapor

enum AlbumFilter: String, Decodable {
    case album
    case compilation
    case single
    case ep
    
    
    
    // MARK: - Properties
    
    var types: [String] {
        switch self {
        case .album:
            return ["Album", "LP"]
        case .compilation:
            return ["Compilation", "Mixed"]
        case .single:
            return ["Single", "Maxi-Single"]
        case .ep:
            return ["EP"]
        }
    }
}

File: FinchServer/Controllers/Utilities/AlbumFilter.cs
namespace FinchServer.Controllers.Utilities;

public enum AlbumFilter {
    Album,
    Compilation,
    Single,
    Ep
}

public static class AlbumFilterExtensions {
    
    // - Functions

    public static string[] Types(this AlbumFilter filter) {
        return filter switch {
            AlbumFilter.Album => ["Album", "LP"],
            AlbumFilter.Compilation => ["Compilation", "Mixed"],
            AlbumFilter.Single => ["Single", "Maxi-Single"],
            AlbumFilter.Ep => ["EP"],
            _ => []
        };
    }
}